### PR TITLE
feat(pipeline): not create subtasks when pipeline canceled

### PR DESCRIPTION
**Pull request by sfwn**
#### What this PR does / why we need it:

Do not create subtasks when pipeline canceled to speed up cancel progress.
Before this pr, when you pressed the cancel button, the sub tasks will be created and set status to NoNeedBySystem.

image

#### Which issue(s) this PR fixes:

- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=305667&issueFilter__urlQuery=eyJzdGF0ZXMiOls0NDEyLDQ1MzgsNDQxMyw0NDE0LDQ0MTUsNDQxNl19&issueTable__urlQuery=eyJwYWdlTm8iOjEsICJwYWdlU2l6ZSI6MTB9&iterationID=1174&tab=BUG&type=BUG)

#### Specified Reviewers:

/assign @Effet @chengjoey 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    not create subtasks when pipeline canceled          |
| 🇨🇳 中文    | 流水线取消时不创建后续任务     |

### CODE REVIEW

**Comment by sfwn** [comment]
scene `auto-test-scenes` is ok.
image
image

File: modules/pipeline/services/pipelinesvc/detail.go [hunk 1]
```diff
@@ -121,6 +122,10 @@ func (s *PipelineSvc) Detail(pipelineID uint64) (*apistructs.PipelineDetailDTO,
 					Value: task.Inspect.Events,
 				})
 			}
+			// set analyzed yaml task to NoNeedBySystem to simulate db task's behaviour
+			if p.Status.IsStopByUser() && task.Status == apistructs.PipelineStatusAnalyzed {
+				task.Status = apistructs.PipelineStatusNoNeedBySystem
+			}
```
Comment (chengjoey) [inline modules/pipeline/services/pipelinesvc/detail.go:128]: is `p.Status.IsStopByUser() ` should change to `p.Status.IsEndStatus()`

Comment (sfwn) [reply modules/pipeline/services/pipelinesvc/detail.go:128]: use `IsStopByUser` is ok.

**Review by chengjoey — APPROVED** [review]

**Comment by sfwn** [comment]
auto test plan execution: 
image

auto test scene execution:
image

**Review by chengjoey — APPROVED** [review]

**Pull request merged**